src/lib/helpers/config.ts

```diff
@@ -0,0 +1,70 @@
+import { z, ZodError } from "zod";
+import { parse } from "@std/toml";
+
+const ConfigSchema = z.object({
+    server: z.object({
+        port: z.number().default(Number(Deno.env.get("PORT")) || 8282),
+        host: z.string().default(Deno.env.get("HOST") || "127.0.0.1"),
+        secret_key: z.string().length(16).default(
+            Deno.env.get("SERVER_SECRET_KEY") || "",
+        ),
+        verify_requests: z.boolean().default(false),
+    }).strict().default({}),
+    cache: z.object({
+        enabled: z.boolean().default(true),
+        directory: z.string().default("/var/tmp"),
+    }).strict().default({}),
+    networking: z.object({
+        ump: z.boolean().default(false),
+        proxy: z.string().nullable().default(Deno.env.get("PROXY") || null),
+        fetch: z.object({
+            timeout_ms: z.number().optional(),
+            retry: z.object({
+                enabled: z.boolean(),
+                times: z.number().optional(),
+                initial_debounce: z.number().optional(),
+                debounce_multiplier: z.number().optional(),
+            }).strict().optional(),
+        }).strict().optional(),
+    }).strict().default({}),
+    jobs: z.object({
+        youtube_session: z.object({
+            po_token_enabled: z.boolean().default(true),
+            frequency: z.string().default("*/5 * * * *"),
+        }).strict().default({}),
+    }).strict().default({}),
+    youtube_session: z.object({
+        oauth_enabled: z.boolean().default(false),
+        cookies: z.string().default(""),
+    }).strict().default({}),
+}).strict();
+
+export type Config = z.infer<typeof ConfigSchema>;
+
+export async function parseConfig() {
+    try {
+        const configFileContents = await Deno.readTextFile("config/config.toml")
```

Comment:
Would that be interesting to be able to specify a custom location for that file? Like `CONFIG_LOCATION`. I can see this being useful for people using the companion as a binary.

Reply:
yep, that'd make sense to add. Would make doing docker setups simpler.

Reply:
Gone with CONFIG_FILE as an env variable, that kind of name is common enough in self-hosted stuff. 

![image](https://github.com/user-attachments/assets/0dfe51c5-9fae-434f-8aec-7c4c6fdb44bd)

Works well